shipyard_blueprints/shipyard/shipyard_api/client.py

```diff
@@ -139,33 +188,111 @@ def export_voyages(self, target_path: str):
-            response.raise_for_status()
+            response = self._request(**request_details)
+            response_json = response.json()
+            logger.info(f"Successfully triggered fleet run: {response_json.get('log')}")
+            return response_json
+        except Exception as e:
+            raise ExitCodeException(
+                f"Error trigger fleet {fleet_id} to run: {e}",
+                exit_code=EXIT_CODE_TRIGGER_FLEET_ERROR,
+            ) from e
+
+    def upsert_fleet(self, fleet_yaml: str):
+        """
+        Upserts a fleet using the provided YAML configuration.
+
+        Args:
+            fleet_yaml (str): The YAML configuration for the fleet.
+
+        Returns:
+            requests.Response: The response from the API.
+
+        Raises:
+            ExitCodeException: If an exit code exception occurs.
+        """
+        try:
+            response = self._request(method="PUT",
+                                     url=f"{self.base_url}/projects/{self.project_id}/fleets",
+                                     headers={
+                                         "accept": "application/json",
+                                         "content-type": "application/json",
+                                         "X-Shipyard-API-Key": self.api_key
+                                     },
+                                     data=yaml.dump(fleet_yaml)
+                                     )
+            logger.info("Successfully upserted fleet")
+            return response
+
         except ExitCodeException:
             raise
         except Exception as e:
             raise ExitCodeException(
-                f"Error trigger fleet {fleet_id} to run: {e}",
-                exit_code=EXIT_CODE_LIST_FLEET_RUNS_ERROR,
-            )
-        else:
-            return response.json()
+                f"Error upserting fleet: {e}",
+                exit_code=EXIT_CODE_UNKNOWN_ERROR,
+            ) from e
+
+
+    def get_run_status(self, fleet_id: str, run_id: str):
+        """
+        Get the status of a fleet run.
+
+        Args:
+            fleet_id (str): The ID of the fleet.
+            run_id (str): The ID of the run.
+
+        Returns:
+            str: The status of the fleet run.
+
+        Raises:
+            ExitCodeException: If an exit code exception occurs.
+        """
+
+        response = self._request("GET", f"{self.base_url}/projects/{self.project_id}/fleets/{fleet_id}/fleetruns",
+                                 headers={
+                                     "accept": "application/json",
+                                     "content-type": "application/json",
+                                     "X-Shipyard-API-Key": self.api_key
+                                 }
+                                 )
+
+        log_data = response.content.decode('utf-8')
+
+        df = pandas.read_csv(io.StringIO(log_data))
```

Comment:
You should include an try/except here to handle cases where the `run_id` does not exist in dataframe. If you leave it as is, this will throw an `IndexError: index 0 is out of bounds for axis 0 with size 0`